### PR TITLE
SAML: Support using Subject in AuthnRequest

### PR DESCRIPTION
**Pull request by tung2744**
ref DEV-1730
ref DEV-1731

The feature can be tested with the demo app using the `Subject NameID` field:
https://authgear-demo-saml-sp.pandawork.com/

### CODE REVIEW

**Comment by louischan-oursky** [comment]
build failed.

**Comment by louischan-oursky** [comment]
Conflicted

**Comment by tung2744** [comment]
Fixed, thanks!

File: pkg/lib/oauth/login_hint.go [hunk 1]
```diff
@@ -33,8 +35,35 @@ type LoginHint struct {
...
+			q.Set("email", h.LoginIDEmail)
+		}
+		if h.LoginIDPhone != "" {
+			q.Set("phone", h.LoginIDPhone)
+		}
+		if h.LoginIDUsername != "" {
+			q.Set("username", h.LoginIDUsername)
+		}
+	default:
+		panic(fmt.Errorf("cannot convert login_hint to string with type: %v", h.Type))
```
Comment (louischan-oursky) [inline pkg/lib/oauth/login_hint.go:54]: This String() method should support other type as well. Add test for this method.

File: pkg/util/setutil/set.go [hunk 2]
```diff
@@ -20,6 +20,10 @@ func NewSetFromSlice[A any, B cmp.Ordered](slice []A, f func(a A) B) Set[B] {
 	return out
 }
 
+func NewStringSetFromSlice(slice []string) Set[string] {
+	return NewSetFromSlice(slice, func(s string) string { return s })
+}
+
```
Comment (louischan-oursky) [inline pkg/util/setutil/set.go:26]: This function seems redundant. The caller can use `NewSetFromSlice(strings, Identity)`.

**Review by louischan-oursky — CHANGES_REQUESTED** [review]

**Review by louischan-oursky — APPROVED** [review]

**Pull request merged**